rest/route/generate_routes.go

```diff
@@ -0,0 +1,63 @@
+package route
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/gorilla/mux"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+func getGenerateManager(route string, version int) *RouteManager {
+	h := &generateHandler{}
+	generate := MethodHandler{
+		RequestHandler: h.Handler(),
+		MethodType:     http.MethodPost,
+	}
+
+	generateRoute := RouteManager{
+		Route:   route,
+		Methods: []MethodHandler{generate},
+		Version: version,
+	}
+	return &generateRoute
+}
+
+type generateHandler struct {
+	files  []json.RawMessage
+	taskID string
+}
+
+func (h *generateHandler) Handler() RequestHandler {
+	return &generateHandler{}
+}
+
+func (h *generateHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	var files []json.RawMessage
+	if err := util.ReadJSONInto(r.Body, &files); err != nil {
+		return errors.Wrap(err, "error reading JSON from body")
+	}
+	h.files = files
+	vars := mux.Vars(r)
+	h.taskID = vars["task_id"]
+	// TODO Do we have a checkTask function in rest v2?
```

Comment:
no, I don't think we do. this would be the first agent-api route to use rest/v2/ and I'm inclined to just do it in agent-api...